packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -284,31 +302,60 @@ export class OpenID4VCIClient {
     return response.successBody;
   }
 
-  getCredentialsSupported(restrictToInitiationTypes: boolean): CredentialsSupported {
+  getCredentialsSupported(restrictToInitiationTypes: boolean, supportedType?: string): CredentialSupported[] {
     const credentialsSupported = this.serverMetadata?.openid4vci_metadata?.credentials_supported;
+    /**
+     * the following (not array part is a legacy code from version 1_0-08 which jff implementors used)
+     */
+    if (credentialsSupported && !Array.isArray(credentialsSupported)) {
+      if (!restrictToInitiationTypes) {
+        return credentialsSupported;
+      }
+      const credentialsSupportedV8: CredentialSupportedV1_0_08 = credentialsSupported as CredentialSupportedV1_0_08;
+      const initiationTypes = this.getCredentialTypes();
+      const supported: IssuerCredentialSubject = {};
+      for (const [key, value] of Object.entries(credentialsSupportedV8)) {
+        if (initiationTypes.includes(key)) {
+          supported[key] = value;
+        }
+      }
+      // todo: fix this later. we're returning CredentialSupportedV1_0_08 as a list of CredentialSupported (for v09 onward)
+      return supported as unknown as CredentialSupported[];
+    }
     if (!credentialsSupported) {
```

Comment:
Suggest to move this to the top. It removes the need for the credentialsSupported && and you are doing the same thing for v8 as well as v8+

Reply:
moved at the top of the method and removed the extra checks